### PR TITLE
Add 'simple' time manager

### PR DESCRIPTION
**Pull request by trre123**
Seems to be at least on par with legacy, most likely better according to my tests. Also far more simple, with less code.

Tuning was done in 2 stages, versus stockfish. 

1. base-pct and ply-pct tuned together at 14s+0s
2. time-factor tuned at 6s+0.1s

Opening-bonus was added as an afterthought and so wasnt tested/tuned.

Test results gotten with these params - (base-pct=2.4,ply-pct=0.0312,time-factor=37.0,opening-bonus=0.0). 

Note the time controls are high enough that legacy tm shouldnt be negatively affected by spf threshold

15s+0s
```
   # PLAYER           :  RATING  ERROR  POINTS  PLAYED   (%)  CFS(%)    W     D    L  D(%)
   1 lc0-simple_tm    :     5.1    8.2   761.0    1500    51      89  185  1152  163    77
   2 lc0-legacy_tm    :     0.0   ----   739.0    1500    49     ---  163  1152  185    77
```
18s+0.3s
```
   # PLAYER           :  RATING  ERROR  POINTS  PLAYED   (%)  CFS(%)    W    D    L  D(%)
   1 lc0-simple_tm    :     8.4    8.9   384.0     750    51      97   60  648   42    86
   2 lc0-legacy_tm    :     0.0   ----   366.0     750    49     ---   42  648   60    86
```

### CODE REVIEW

**Comment by trre123** [comment]
Updated defaults and ran new tests including the param 'opening-bonus'

15s+0s
```
   # PLAYER        :  RATING  ERROR  POINTS  PLAYED   (%)  CFS(%)    W     D    L  D(%)
   1 lc0-simple    :    10.8    8.5   773.0    1500    52      99  209  1128  163    75
   2 lc0-legacy    :     0.0   ----   727.0    1500    48     ---  163  1128  209    75
```
18s+0.3s
```
   # PLAYER        :  RATING  ERROR  POINTS  PLAYED   (%)  CFS(%)    W    D    L  D(%)
   1 lc0-simple    :     9.8    9.6   385.5     750    51      98   69  633   48    84
   2 lc0-legacy    :     0.0   ----   364.5     750    49     ---   48  633   69    84
```

**Comment by trre123** [comment]
Also ran tests against smooth tm (pr1762)

15s+0s
```
   # PLAYER        :  RATING  ERROR  POINTS  PLAYED   (%)  CFS(%)    W    D    L  D(%)
   1 lc0-simple    :    36.2   12.6   413.5     750    55     100  140  547   63    73
   2 lc0-smooth    :     0.0   ----   336.5     750    45     ---   63  547  140    73
```
18s+0.3s
```
   # PLAYER        :  RATING  ERROR  POINTS  PLAYED   (%)  CFS(%)    W    D    L  D(%)
   1 lc0-simple    :    14.8   12.5   260.5     500    52      99   55  411   34    82
   2 lc0-smooth    :     0.0   ----   239.5     500    48     ---   34  411   55    82
```

**Comment by trre123** [comment]
Gauntlet against SF @ 18s+0.3s
```
   # PLAYER        :  RATING  ERROR  POINTS  PLAYED   (%)  CFS(%)     W     D    L  D(%)
   1 sf            :     0.0   ----  2640.0    5000    53     100  1147  2986  867    60
   2 lc0-simple    :   -17.9    8.6  1186.5    2500    47      73   444  1485  571    59
   3 lc0-legacy    :   -21.6    8.4  1173.5    2500    47     ---   423  1501  576    60
```

**Comment by zz4032** [comment]
I don't think `simple` is a suitable name here when another TM `alphazero` has only 1 parameter instead of 4 and 20 lines less of code.

**Comment by trre123** [comment]
> I don't think `simple` is a suitable name here when another TM `alphazero` has only 1 parameter instead of 4 and 47 lines less of code.

Yea but alphazero also sucks and shouldnt even be considered a real tm. It completely falls apart when attempting to go between different time controls while keeping the same param values.

I suppose I could rename this PR to 'the-simple-yet-relevant' tm? Rolls right off the tongue doesnt it?

File: src/mcts/stoppers/simple.cc [hunk 1]
```diff
@@ -0,0 +1,123 @@
+/*
+  This file is part of Leela Chess Zero.
+  Copyright (C) 2020 The LCZero Authors
```
Comment (borg323) [inline src/mcts/stoppers/simple.cc:3]: 2022

File: src/mcts/stoppers/simple.cc [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+class SimpleTimeManager : public TimeManager {
+ public:
+  SimpleTimeManager(int64_t move_overhead, const OptionsDict& params)
+      : move_overhead_(move_overhead),
+        basepct_(params.GetOrDefault<float>("base-pct", 1.83f)),
+        plypct_(params.GetOrDefault<float>("ply-pct", 0.0454f)),
+        timefactor_(params.GetOrDefault<float>("time-factor", 33.4f)),
+        opening_bonus_(params.GetOrDefault<float>("opening-bonus", 82.5f)) {
+    if (basepct_ <= 0.0f || basepct_ > 100.0f)
```
Comment (borg323) [inline src/mcts/stoppers/simple.cc:42]: Use braces when the if statement is more than one line.

File: meson.build [hunk 3]
```diff
@@ -166,6 +166,7 @@ files += [
   'src/mcts/node.cc',
   'src/mcts/params.cc',
   'src/mcts/search.cc',
+  'src/mcts/stoppers/simple.cc',
```
Comment (borg323) [inline meson.build:169]: Move it further down so the files are in alphabetical order.

**Review by borg323 — COMMENTED** [review]
Just style comments, clang-format does a few more small changes.

**Comment by borg323** [comment]
I don't mind the name, it is simple enough.

**Comment by trre123** [comment]
> Just style comments, clang-format does a few more small changes.

Done, thanks

**Review by borg323 — APPROVED** [review]

**Pull request merged**